src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -325,6 +350,19 @@ void remove(int filteredIndex) throws IllegalValueException {
             }
         }
 
+        void removeAll(Collection<E> toRemove) {
+            for (E item : toRemove) {
+                if (filteredList.contains(item)) {
```

Comment:
Hmph, this is doing an `O(n)` search for each `toRemove`.

Wouldn't we get the exact same results if we did a `removeAll(Predicate<? super E>)` which will filter the filteredList?